libs/kameleon/src/kameleon/app_listing.clj

```diff
@@ -44,6 +44,19 @@
               (where {:app_category_app.app_category_id
                       fav_group_id_subselect})))))
 
+(defn- add-app-id-where-clause
+  [query {:keys [app-ids]}]
+  (if (seq app-ids)
+    (where query {:id [in app-ids]})
+    query))
+
+(defn- add-agave-pipeline-where-clause
+  [query {agave-enabled? :agave-enabled :or {agave-enaled? "false"}}]
+  (let [agave-enabled? (Boolean/parseBoolean agave-enabled?)]
+    (if-not agave-enabled?
+      (where query {:step_count :task_count})
+      query)))
```

Comment:
Moved from `kameleon.queries` and made private because it wasn't called from anywhere else.